### ISSUE TITLE
local MathJax work weird with v4.1.0

### ISSUE THREAD

**Issue by g4gilson** [asciidoctor/asciidoctor-reveal.js#408]
I try to use MathJax with a local configuration:
```
:stem:
:mathjaxdir: local/MathJax-2.7.6
```
but I get a strange rendering:
![image](https://user-images.githubusercontent.com/63807744/102782544-d0f97a00-4399-11eb-9c7b-2361b83b0519.png)

I tried with Mathjax v2.4.0, v2.7.5 and v2.7.6.

In html I get a link like:
local/MathJax-2.7.6/MathJax.js?config=TeX-MML-AM_HTMLorMML

when I modify it by: 
local/MathJax-2.7.6/MathJax.js?config=TeX-MML-AM_CHTML
I get this:
![image](https://user-images.githubusercontent.com/63807744/102783755-b88a5f00-439b-11eb-951c-ea0f1ceef275.png)

So, I don't know what change between these configuration but, in local and online (I do not clear cache) I get a good render with:
config=TeX-MML-AM_CHTML 
configuration.

**Comment by Mogztter** [comment]
Are you using a local HTTP server to preview the presentation? Any errors in the Web console?
Also, which browser are you using?

For reference, we are using the same config as the built-in HTML5 converter: https://docs.mathjax.org/en/v2.7-latest/config-files.html#the-am-chtml-configuration-file

On the browser extension, I'm already using the CHTML output (with MathJax 3). 

> The CommonHTML output processor renders your mathematics using HTML with CSS styling. It produces high-quality output in all modern browsers, with results that are consistent across browsers and operating systems. **This is MathJax’s primary output mode since MathJax version 2.6.** Its major advantage is its quality, consistency, and the fact that its output is independent of the browser, operating system, and user environment. This means you can pre-process mathematics on a server, without needing to know the browser, what fonts are available, and so on. **(In version 2, both the HTML-CSS and NativeMML processors produced different output for different browsers and user environments.)**

http://docs.mathjax.org/en/latest/output/html.html#html-support

So we might consider switching to CommonHTML but first I want to understand why you are experiencing this issue (or rather under what conditions this happens).

**Comment by g4gilson** [comment]
I have no error in console and I haven't using a local server, I just open the html with my navigator.
In firefox 84.0 inspector:
When the page work, config=TeX-MML-AM_CHTML, I have in the html to print the ≈ :
```
<span id="MJXc-Node-47" class="mjx-mo" style="padding-left: 0.333em; padding-right: 0.333em;">
<span class="mjx-char MJXc-TeX-main-R" style="padding-top: 0.201em; padding-bottom: 0.313em;">≈
</span>
</span>
```
When the page doesn't work, config=TeX-MML-AM_HTMLorMML, I have in the html to print the ≈ :
```
<span class="mo" id="MathJax-Span-47" style="padding-left: 0.333em; padding-right: 0.333em;">
<img src="file:///local/MathJax-2.7.6/fonts/HTML-CSS/TeX/png/Main/Regular/476/2248.png?V=2.7.6" style="width: 34px; height: 20px; vertical-align: 2px; margin-right: 0.056em;">
</span>
```
file:///local/MathJax-2.7.6/fonts/HTML-CSS/TeX/png/Main/Regular/476/2248.png is this image:
![image](https://user-images.githubusercontent.com/63807744/102890026-06669c00-445c-11eb-8363-7e716c093841.png)

When I use the same configuration but instead of use local link, I use :
https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.6/MathJax.js?config=TeX-MML-AM_HTMLorMML
I have this in html:
```
<span class="mo" id="MathJax-Span-47" style="font-family: MathJax_Main; padding-left: 0.333em; padding-right: 0.333em;">≈</span>
```
and it is printed properly. I don't know how it is possible, I download the MathJax v2.7.6 to use it locally here:
https://github.com/mathjax/MathJax/releases/tag/2.7.6

**Comment by Mogztter** [comment]
I cannot reproduce this issue, I just did a test with MathJax 2.7.6 and it's working fine in both Chrome and Firefox (with or without a local HTTP server).

I don't know why but it seems that in your case MathJax is falling back to images instead of HTML + CSS. Since we apply a style on all `<img>` elements (most specifically a black border) it does not look so great.


### Chrome

![chrome](https://user-images.githubusercontent.com/333276/102905515-9106c580-4473-11eb-8735-a333b56f8818.png)

### Firefox

![firefox](https://user-images.githubusercontent.com/333276/102905521-92d08900-4473-11eb-8d44-196c1c3e04a9.png)

**Comment by g4gilson** [comment]
to reproduce the error:
1) download MathJax v2.7.6 to use it locally here: https://github.com/mathjax/MathJax/releases/tag/2.7.6
2) In examples/mathjax-cdn.adoc replace the 
```
:mathjaxdir: [...]
```
by
```
:mathjaxdir: ../MathJax-2.7.6
```
and place the download MathJax dir properly
3) convert it with ascii doctor reveal js v4.1.0 and print it in a navigator
I have that:
![image](https://user-images.githubusercontent.com/63807744/103003163-9118cb00-4530-11eb-9f32-832f3afa4d72.png)
4) replace again the same line by 
```
:mathjaxdir: MathJax-2.7.6
```
and place the download MathJax dir properly
5) convert it with ascii doctor reveal js v4.1.0 and print it in a navigator
I have that:
![image](https://user-images.githubusercontent.com/63807744/103003335-e48b1900-4530-11eb-98b2-510873939c45.png)

the relative path with .. seems to not work properly

**Comment by Mogztter** [comment]
@g4gilson I can only reproduce this issue using Firefox without a local HTTP server. On Chrome it's working even when I'm not using a local HTTP server.

Anyway, I think it's a limitation/issue on MathJax.

We can keep this issue open to make sure that this issue is gone once we've upgraded to MathJax 3: https://github.com/asciidoctor/asciidoctor-reveal.js/issues/369

**Comment by g4gilson** [comment]
Nice, I find a way that can be reproduced. Thanks for your answers :+1:.
So if I understand, I can fix it with a local HTTP server. How can I setup that? I can launch one reveal js http server on port: 18736 but how can I configure it in .adoc to get it?

**Comment by Mogztter** [comment]
You need to convert your AsciiDoc presentation to HTML and then serve this HTML file using a local HTTP server.

Depending on your environment you have several options. I guess using Python `SimpleHTTPServer` is probably the easiest solution:

```
python -m SimpleHTTPServer
```

You will need to serve from the root, otherwise `..` won't be resolved. In other words, if you have the following directory structure, you will need to start the local HTTP server at the root (not in the `docs` directory):

```
.
├── docs
│   ├── presentation.adoc
│   └── presentation.html
└── mathajax-2.7.6
```

And use `http://localhost:8000/docs/presentation.html` to preview your presentation.

**Comment by g4gilson** [comment]
thank you, that works as expected :+1:

**Comment by archiloque** [comment]
@Mogztter as the update to MathJax 3 has been done, shall this issue be closed ?

**Comment by ggrossetie** [comment]
Following up on the last question here — yes, this is resolved. `main` has used MathJax 3 (currently pinned to 3.2.0) as the default for a while now, which doesn't have the MathJax 2.x HTML-CSS output's font-image fallback behavior that caused the broken rendering in this thread (the `file://` without a local server + Firefox combination).

Closing as resolved by the MathJax 3 migration (#369).
